extensions/wikia/UserRenameTool/process/ProcessBase.class.php

```diff
@@ -0,0 +1,556 @@
+	 * @param string $reason
+	 * @param bool $notifyUser Whether to notify the renamed user
+	 */
+	public function __construct( $oldUsername, $newUsername, $confirmed = false, $reason = null, $notifyUser = true ) {
+		global $wgUser;
+
+		$this->startTime = time();
+
+		// Save original request data
+		$this->requestData = new \stdClass();
+		$this->requestData->oldUsername = $oldUsername;
+		$this->requestData->newUsername = $newUsername;
+
+		$this->actionConfirmed = $confirmed;
+		$this->reason = $reason;
+		$this->requestorId = $wgUser ? $wgUser->getId() : 0;
+		$this->requestorName = $wgUser ? $wgUser->getName() : '';
+		$this->notifyUser = $notifyUser;
+	}
+
+	/**
+	 * Create a new UserRenameToolProcessLocal object from select context data.
+	 *
+	 * @param array $data
+	 * @return ProcessBase
+	 */
+	public static function newFromData( $data ) {
+		$o = new static( $data['rename_old_name'], $data['rename_new_name'], '', true );
+
+		$mapping = [
+			'userId' => 'rename_user_id',
+			'oldUsername' => 'rename_old_name',
+			'newUsername' => 'rename_new_name',
+			'fakeUserId' => 'rename_fake_user_id',
+			'requestorId' => 'requestor_id',
+			'requestorName' => 'requestor_name',
+			'phalanxBlockId' => 'phalanx_block_id',
+			'reason' => 'reason',
+			'renameIP' => 'rename_ip',
+			'startTime' => 'start_time',
+		];
+
+		foreach ( $mapping as $property => $key ) {
+			if ( array_key_exists( $key, $data ) ) {
+				$o->$property = $data[$key];
+			}
+		}
+
+		// Quick hack to recover requestor name from its id
+		if ( !empty( $o->requestorId ) && empty( $o->requestorName ) ) {
+			$requestor = \User::newFromId( $o->requestorId );
+			$o->requestorName = $requestor->getName();
+		}
+
+		$o->logDebug( "Created new ProcessBase instance in newFromData()" );
+
+		return $o;
+	}
+
+	public function getErrors() {
+		return $this->errors;
+	}
+
+	public function getWarnings() {
+		return $this->warnings;
+	}
+
+	public function getCurrentTaskId() {
+		return $this->currentTaskId;
+	}
+
+	/**
+	 * Saves passed error for future retrieval
+	 *
+	 * @param string $msg Error message
+	 */
+	protected function addError( $msg ) {
+		$this->errors[] = $msg;
+	}
+
+	/**
+	 * Saves passed warning for future retrieval
+	 *
+	 * @param string $msg Error message
+	 */
+	protected function addWarning( $msg ) {
+		$this->warnings[] = $msg;
+	}
+
+	/**
+	 * Rename the user in the table given.
+	 *
+	 * @param \DatabaseBase $dbw Database to operate on
+	 * @param string $table Table name
+	 * @param int $uid User ID
+	 * @param string $oldUserName Old username
+	 * @param string $newUserName New username
+	 * @param array $extra Extra options (currently: userid_column, username_column, conds)
+	 *
+	 * @return bool
+	 */
+	public function renameInTable( $dbw, $table, $uid, $oldUserName, $newUserName, $extra ) {
+		$dbName = $dbw->getDBname();
+		$this->logInfo( sprintf( "Processing %s.%s.%s", $dbName, $table, $extra['username_column'] ) );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "username_column" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)